presto-main/src/main/java/io/prestosql/operator/exchange/LocalExchangeSource.java

```diff
@@ -19,41 +19,37 @@
 import io.prestosql.operator.WorkProcessor.ProcessState;
 import io.prestosql.spi.Page;
 
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.ThreadSafe;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Consumer;
 
 import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
 import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
 public class LocalExchangeSource
 {
-    private static final SettableFuture<?> NOT_EMPTY;
-
-    static {
-        NOT_EMPTY = SettableFuture.create();
-        NOT_EMPTY.set(null);
-    }
+    private static final ListenableFuture<?> NOT_BLOCKED = immediateFuture(null);
 
     private final Consumer<LocalExchangeSource> onFinish;
 
-    private final BlockingQueue<PageReference> buffer = new LinkedBlockingDeque<>();
+    private final LinkedBlockingDeque<PageReference> buffer = new LinkedBlockingDeque<>();
```

Comment:
Why do we need to change BlockingQueue to LinkedBlockingDeque?

Reply:
Accidentally left this behind, I changed it because it was easier to navigate into the implementation of `LinkedBlockingDeque` methods in my IDE if the field was declared with the exact type. Will revert it to the previous declaration.